File: EaiGateway/src/main/java/com/yung/cho/eaigateway/logging/GatewayLogProducer.java
package com.yung.cho.eaigateway.logging;

import lombok.RequiredArgsConstructor;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class GatewayLogProducer {

	private final KafkaTemplate<String, byte[]> kafkaTemplate;
	
	public void send(String topic, byte[] message) {
		kafkaTemplate.send(topic, message);		// fire and forget...!! 비동기식 publishing
	}
}

File: EaiGateway/src/main/kotlin/com/yung/cho/eaigateway/logging/GatewayLogProducer.kt
package com.yung.cho.eaigateway.logging

import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service

@Service
class GatewayLogProducer(
    private val kafkaTemplate: KafkaTemplate<String, ByteArray>
) {
    fun send(topic: String, message: ByteArray) {
        kafkaTemplate.send(topic, message) // fire and forget...!! 비동기식 publishing
    }
}
